### ISSUE TITLE
Llama-3.2-11B-Vision (mllama): errors with PagedAttention by default, and image inference crashes on CUDA with `--paged-attn off`

### ISSUE THREAD

**Issue by subin9** [EricLBuehler/mistral.rs#2335]
## Describe the bug

Running `meta-llama/Llama-3.2-11B-Vision-Instruct` (or the `unsloth/` mirror) — the `vllama`/mllama architecture — is broken end-to-end for image inputs:

**1. Default launch fails immediately.** On CUDA, PagedAttention is enabled by default, and the model aborts loading with:

```
Error: Expected eager attention implementation
```

This comes from `MLlamaTextModel::new` (`mistralrs-core/src/vision_models/mllama/text.rs`), which hard-requires eager attention. The vision loader, however, advertises paged support:

```rust
// mistralrs-core/src/pipeline/loaders/multimodal_loaders.rs  (VLlamaLoader)
fn supports_paged_attention(&self, _config: &str) -> bool { true }
```

so the multimodal pipeline never auto-disables PagedAttention (unlike Qwen2-VL / Qwen2.5-VL, whose loaders return `false` and therefore fall back to eager cleanly). Same failure shape as #1692 (Qwen3 Embedding).

**2. With `--paged-attn off`, text works but images crash.** The model now loads (eager), text generation works, but the first image request crashes:

```
DriverError(CUDA_ERROR_INVALID_VALUE, "invalid argument")
  ...
  2: <candle_core::cuda_backend::CudaStorage as ...>::to_dtype
  3: candle_core::tensor::Tensor::to_dtype
  4: <mistralrs_core::vision_models::mllama::MLlamaModel as ...>::forward
```

Note this crash reproduces **only on CUDA, not on CPU** — which is why text (and CPU) appears to work.

## Root cause (bug 2)

`VLlamaLoader::supports_paged_attention` returns `true` while `MLlamaTextModel` only supports eager. It should return `false` so PagedAttention is auto-disabled for this model, matching the Qwen2-VL loaders.

## Root cause (image crash)

On **decode** steps the mllama inputs processor still runs its image block, because `Sequence::take_images()` clones (rather than empties) the images when `has_changed_prompt` is false, so `has_images()` stays true after prefill. On a decode step the current input has no `<|image|>` token, so `convert_sparse_cross_attention_mask_to_dense` produces a cross-attention mask with a zero-sized `max_num_images` dimension, e.g. `[1, 1, 0, 4]` (0 elements). `prepare_cross_attention_mask` then calls `.to_dtype(F32)` on that 0-element tensor. On CUDA, launching a kernel over 0 elements returns `CUDA_ERROR_INVALID_VALUE`; on CPU it is a harmless no-op (hence text/CPU work).

Instrumented trace (tiny repro model, CUDA):

```
# prefill (ok):
forward_inner: pixel_values=Some([1,1,4,3,560,560]) image_hashes.len=1 cross_attn_mask=Some(([1,44,1,4], I64))
# decode (crash):
forward_inner: pixel_values=Some([1,1,4,3,560,560]) image_hashes.len=0 cross_attn_mask=Some(([1,1,0,4], I64))  <- 0 elements
-> to_dtype(F32) on a 0-element CUDA tensor -> CUDA_ERROR_INVALID_VALUE
```

`Qwen2VLImageProcessor` avoids this by nulling image inputs on non-prompt steps (`mistralrs-core/src/vision_models/qwen2vl/inputs_processor.rs`: `let mut pixel_values = if is_prompt { pixel_values } else { None };`). mllama has no equivalent guard. Cross-attention is skipped during decode anyway (`mllama/text.rs`: `if cross_attn_states.is_none() { continue }`), so passing image inputs on decode is both wasteful and, on CUDA, fatal. Possibly related to #868 (interactive mode reusing the first image — same `take_images`/`has_changed_prompt` area).

## Additional finding (vision head config not read)

`MLlamaVisionConfig::num_attention_heads` is deserialized with `#[serde(default = "d_attn_heads")]` (=16) but **without** an alias for HF's actual key `attention_heads`, so the vision head count from `config.json` is ignored and always defaults to 16. The stock 11B happens to have 16 vision heads, so it is masked; any mllama checkpoint with a different vision head count silently loads with the wrong dimensions (or fails with a weight shape mismatch). HF's `MllamaVisionConfig` field is `attention_heads`.

## Minimum reproducible example

Because this is architecture-level, a tiny random-weight mllama reproduces it on GPU in seconds (no 20 GB download):

```bash
# ~22 MB, MllamaForConditionalGeneration, cross_attention_layers=[1]
MODEL=yujiepan/llama-3.2-vision-tiny-random

# 1) default -> "Expected eager attention implementation"
cargo run --release --features cuda -- serve -p 1234 auto -m $MODEL

# 2) eager -> loads; text ok; first image -> CUDA_ERROR_INVALID_VALUE
cargo run --release --features cuda -- serve -p 1234 --no-ui --paged-attn off auto -m $MODEL
curl -s localhost:1234/v1/chat/completions -H 'Content-Type: application/json' -d '{
  "model":"default","max_tokens":8,
  "messages":[{"role":"user","content":[
    {"type":"text","text":"What is in this image?"},
    {"type":"image_url","image_url":{"url":"data:image/png;base64,<any small png>"}}
  ]}]}'
```

(The tiny model's `config.json` uses `attention_heads: 2`, which also triggers the "Additional finding" load failure until the alias is added.)

The stock `meta-llama/Llama-3.2-11B-Vision-Instruct` reproduces items 1 and 2 identically on CUDA.

## Other information

- OS: Linux (WSL2)
- GPU: NVIDIA RTX 4070 Ti SUPER (16 GB); CUDA 12.9 (`nvcc` V12.9.86)
- Built with `--features cuda`

## Latest commit or version

v0.8.23 / current `master`. All three findings are in unmodified mllama code paths.

---

I have a fix ready (loader `supports_paged_attention` -> false; gate the mllama image block on `is_prompt`; add `#[serde(alias = "attention_heads")]`) and will open a PR referencing this issue.
